src/codegen/c.zig

```diff
@@ -83,6 +98,28 @@ pub fn generate(file: *C, decl: *Decl) !void {
     }
 }
 
+pub fn generateHeader(
+    arena: *std.heap.ArenaAllocator,
+    module: *Module,
+    header_info: *HeaderInfo,
+    decl: *Decl,
+) !void {
+    switch (decl.typed_value.most_recent.typed_value.ty.zigTypeTag()) {
+        .Fn => {
+            var error_msg: *ErrorMsg = undefined;
+            const writer = header_info.buf.writer();
+            renderFunctionSignature(arena, &error_msg, header_info, writer, decl) catch |err| {
+                if (err == error.AnalysisFail) {
```

Comment:
This also means that the error_msg is never used, which means you can probably remove it as a parameter and all related logic.

Reply:
Yeah, I was only doing this because TODOs return `error.AnalysisFail`. If we're saying that we can just replace those with panics (which makes sense, now that I think about it), we can get rid of this.

Reply:
On second thought, leave it alone. Normal codegen uses AnalysisFail for TODOs; we should follow it. This should be fine, ignore my initial comments.